apps/common/components/Table.tsx

```diff
@@ -0,0 +1,97 @@
+import {useCallback, useMemo, useState} from 'react';
+import {sort} from '@veYFI/utils';
+import IconChevronPlain from '@common/icons/IconChevronPlain';
+
+import type {ReactElement} from 'react';
+
+type TSortOrder = 'asc' | 'desc' | '';
+
+type TState<T> = {
+	sortedBy: Extract<keyof T, string> | undefined, 
+	order: TSortOrder
+}
+
+const switchOrder = (order: TSortOrder): TSortOrder => order === '' ? 'desc' : order === 'desc' ? 'asc' : '';
+
+type TMetadata<T> = {
+	key: Extract<keyof T, string>;
+	label: string;
+	className?: string;
+	sortable?: boolean;
+	fullWidth?: boolean;
+	columnSpan?: number;
+	format?: (item: T) => string;
+	transform?: (item: T) => ReactElement;
+}
+  
+type TTableProps<T> = {
+	metadata: TMetadata<T>[];
+	data: T[];
+	columns?: number;
```

Comment:
Perhaps a philosophical question here, but is it a table if there're no columns?

Reply:
Sorry my question wasn't clear, I meant if the `columns` should be required :) (as it's currently optional)

Reply:
Ah I see, ok makes sense!